src/webapp/components/theme-list/ThemeListTable.tsx

```diff
@@ -32,20 +37,48 @@ export interface ThemeDetail {
     subtitle: string;
     logo: ReactNode;
     palette: string[];
+    users: SharingRule[];
+    userGroups: SharingRule[];
 }
 
 type ThemeListTableProps = Pick<RouteComponentProps, "themes" | "setThemes">;
 
 export default function ThemeListTable({ themes, setThemes }: ThemeListTableProps) {
-    const { compositionRoot } = useAppContext();
+    const { api, compositionRoot } = useAppContext();
     const snackbar = useSnackbar();
     const loading = useLoading();
 
     const [selection, setSelection] = useState<TableSelection[]>([]);
     const [themeEdit, setThemeEdit] = useState<{ type: "edit" | "new"; theme?: Theme }>();
+
+    const [currentUser, setCurrentUser] = useState<User>();
+    getCurrentUser(api).then(user => setCurrentUser(user));
+
     const [warningDialog, setWarningDialog] = useState<WarningDialog | null>(null);
 
     const rows = buildThemeDetails(themes);
+    const usersVisibility = rows
+        .map(row => row.users)
+        .map(user => user.map(u => u.id))
+        .map(u => {
+            const arr = u.map(id => {
+                return currentUser?.id === id;
+            });
+            return arr.every(Boolean) && arr.length !== 0;
+        });
```

Comment:
This results in a variable of type `boolean[]`, to be indexed by position. As a rule of thumb, using indexes is a red flag, we should build a proper mapping (typicall IDs as keys). Lodash `fromPairs` is the key method to build theses mappings. A proposal (untested, make sure it works): 

```
const usersVisibility = _(rows)
        .map((themeDetail): [Id, boolean] => {
            const isCurrentUserInUsers = _(themeDetail.users).some(user => user.id === currentUser?.id);
            return [themeDetail.id, isCurrentUserInUsers];
        })
        .fromPairs()
        .value();
```